pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -71,6 +71,21 @@ type MonitoringStackSpec struct {
 	// +optional
 	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
 
+	// CreateClusterRoleBinding for a Monitoring Stack Resource
+	// If a NamespaceSelector is given, the controller can create a
+	// ClusterRoleBinding for the Prometheus and Alertmanager
+	// ServiceAccounts. This allows the
+	// ServiceAccount access to all namespaces, allowing the
+	// ServiceDiscovery to work across namespaces out of the
+	// box. However by impersonating this ServiceAccount a user could elevate
+	// their access in unintended ways.
+	// To avoid this set CreateClusterRoleBinding to false. Note
+	// that admins must create the needed namespaced RoleBindings manually
+	// so that endpoint discovery works as expected.
+	// +kubebuilder:default="true"
+	// +optional
+	CreateClusterRoleBinding bool `json:"createClusterRoleBinding,omitempty"`
```

Comment:
Kube API conventions discourage the use of boolean fields and suggest enum values instead.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Reply:
Thanks, good point. What do you think  of the naming?